pkg/config/spec.go

```diff
@@ -722,3 +698,27 @@ func isMounted(config *Config, part *v1.Partition) (bool, error) {
 	}
 	return !notMnt, nil
 }
+
+func setSourceSize(cfg *Config, spec *v1.UpgradeSpec) error {
+	var size int64
+	var err error
+
+	if spec.RecoveryUpgrade {
+		size, err = GetSourceSize(cfg, spec.Recovery.Source)
+	} else {
+		size, err = GetSourceSize(cfg, spec.Active.Source)
+	}
```

Comment:
I think in this cases it's better to use the control flow to resturn a temp variable than to call the function on both branches

```suggestion
        var source whatverTypeSourceIs
        
	if spec.RecoveryUpgrade {
		source = spec.Recovery.Source
	} else {
		source = spec.Active.Source
	}
	
	size, err = GetSourceSize(cfg, source)
```

Reply:
Actually our first take was to store `spec.Recovery` or `spec.Active` to a variable so that we can use it below when setting the Size too. But for some reason it didn't work (pointers in Go and such). I'm actually curious to understand why so I'll give it another shot.

Reply:
Code has been refactored. This no longer applies.